packages/js/product-editor/src/components/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -133,6 +120,55 @@ export const AttributeTermInputField: React.FC<
 		}
 	};
 
+	const createAttributeTerm = async (
+		attribute: Partial< ProductAttributeTerm >
+	) => {
+		recordEvent( 'product_attribute_term_add', {
+			source: TRACKS_SOURCE,
+		} );
+		setIsCreatingTerm( true );
+		try {
+			const newAttribute: ProductAttributeTerm =
+				await createProductAttributeTerm( {
+					...attribute,
+					attribute_id: attributeId,
+				} );
+			recordEvent( 'product_attribute_term_add_success', {
+				source: TRACKS_SOURCE,
+			} );
+			onChange( [ ...value, newAttribute ] );
+			invalidateResolutionForStoreSelector( 'getProductAttributes' );
+			setIsCreatingTerm( false );
+		} catch ( e ) {
+			recordEvent( 'product_attribute_term_add_failed', {
+				source: TRACKS_SOURCE,
+			} );
+			createNotice(
+				'error',
+				__( 'Failed to create attribute term.', 'woocommerce' )
```

Comment:
We may want to also support the `term_exists` error code and mention the same attribute term name already exists.
This may never happen in our UI, only if something is broken I suppose and we allow users to create a term that already exists.

Reply:
Good eyes! I addressed this in the commit [3aae94c](https://github.com/woocommerce/woocommerce/pull/39610/commits/3aae94c2ee3631a349d3429e4c7ed17943162825)